javatests/arcs/android/integration/policy/PolicyTest.kt

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2021 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+package arcs.android.integration.policy
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import arcs.android.integration.IntegrationEnvironment
+import arcs.core.data.RawEntity
+import arcs.core.host.toRegistration
+import arcs.core.storage.referencemode.ReferenceModeStorageKey
+import arcs.core.util.testutil.LogRule
+import com.google.common.truth.Truth.assertThat
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.withTimeout
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@OptIn(ExperimentalCoroutinesApi::class, kotlin.time.ExperimentalTime::class)
+@RunWith(AndroidJUnit4::class)
+class PolicyTest {
+
+  @get:Rule
+  val log = LogRule()
+
+  /** Note, a new IntegrationEnvironment is allocated for each test method. */
+  @get:Rule
+  val env = IntegrationEnvironment()
+
+  /**
+   * Scenario: A policy-compliant recipe with @persistent handles egresses data
+   * and stores ingress-restricted values to disk that is never deleted.
+   */
+  @Test
+  fun persistentHandlesWithEgress_StoresIngressRestrictedValues_neverDeleted() = runBlocking {
+    env.addNewHostWith(
+      ::IngressThing.toRegistration(),
+      ::EgressAB.toRegistration()
+    )
+
+    val arc = env.startArc(PersistsEgressesPlan)
```

Comment:
Mostly a clarifying question: where do you get `PersistsEgressesPlan`?

Reply:
This is code-generated for us via `arcs_kt_gen`.